packages/php/blueprint/src/ImportStep.php

```diff
@@ -80,33 +82,80 @@ function ( $key, $importer ) {
 	public function import() {
 		$result = StepProcessorResult::success( $this->step_definition->step );
 
-		if ( ! isset( $this->indexed_importers[ $this->step_definition->step ] ) ) {
-			$result->add_warn( "Unable to find an importer for {$this->step_definition->step}" );
+		if ( ! $this->can_import( $result ) ) {
 			return $result;
 		}
 
 		$importer = $this->indexed_importers[ $this->step_definition->step ];
-		// validate importer is a step processor before processing.
+
+		$logger = new Logger();
+		$logger->log(
+			sprintf( 'Starting import "%s" step', $this->step_definition->step ),
+			WC_Log_Levels::INFO,
+			array(
+				'importer' => get_class( $importer ),
+			)
+		);
+
+		$importer_result = $importer->process( $this->step_definition );
+
+		if ( $importer_result->is_success() ) {
+			$logger->log(
+				sprintf( 'Import "%s" step completed', $this->step_definition->step ),
+				WC_Log_Levels::INFO,
+				array(
+					'messages' => $importer_result->get_messages( 'info' ),
+				)
+			);
+		} else {
+			$logger->log(
+				sprintf( 'Import "%s" step failed', $this->step_definition->step ),
+				WC_Log_Levels::ERROR,
+				array(
+					'messages' => $importer_result->get_messages( 'error' ),
+				)
+			);
+		}
+
+		$result->merge_messages( $importer_result );
+
+		return $result;
+	}
+
+	/**
+	 * Check if the step can be imported.
+	 *
+	 * @param StepProcessorResult $result The result object to add messages to.
+	 *
+	 * @return bool True if the step can be imported, false otherwise.
+	 */
+	protected function can_import( &$result ) {
+		// Check if the importer exists.
+		if ( ! isset( $this->indexed_importers[ $this->step_definition->step ] ) ) {
+			$result->add_error( 'Unable to find an importer' );
+			return false;
+		}
+
+		$importer = $this->indexed_importers[ $this->step_definition->step ];
+		// Validate importer is a step processor before processing.
 		if ( ! $importer instanceof StepProcessor ) {
-			$result->add_warn( "Importer {$this->step_definition->step} is not a valid step processor" );
-			return $result;
+			$result->add_error( 'Incorrect importer type' );
```

Comment:
Any specific reason for changing this and the one above to `error` instead of `warn`?

Reply:
I realized that using `warn` would be problematic for this case because it wouldn't be visible in the UI, and the import would be considered successful, but it wouldn't import anything.